tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java

```diff
@@ -2344,16 +2350,74 @@ static Set<String> getPropertySet() {
   public static final String TEZ_TASK_ATTEMPT_HOOKS = TEZ_TASK_PREFIX + "attempt.hooks";
 
   /**
-   * Comma-separated list of additional hadoop config files to load from CLASSPATH in standalone mode.
+   * String value. ZooKeeper quorum connection string used when creating a CuratorFramework for the ZooKeeper registry.
    */
   @ConfigurationScope(Scope.AM)
   @ConfigurationProperty
-  public static final String TEZ_AM_STANDALONE_CONFS = TEZ_AM_PREFIX + "standalone.confs";
+  public static final String TEZ_AM_ZOOKEEPER_QUORUM = TEZ_AM_PREFIX + "zookeeper.quorum";
 
   /**
-  * String value. The class to be used for the AM registry.
-  */
+   * String value. Namespace in ZooKeeper registry for the Application Master.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty
+  public static final String TEZ_AM_REGISTRY_NAMESPACE = TEZ_AM_PREFIX + "registry.namespace";
+  public static final String TEZ_AM_REGISTRY_NAMESPACE_DEFAULT = "/tez_am/server";
+
+  /**
+   * Boolean value. Whether to enable compute groups, see further details in ZkConfig.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty
+  public static final String TEZ_AM_REGISTRY_ENABLE_COMPUTE_GROUPS = TEZ_AM_PREFIX + "registry.enable.compute.groups";
+  public static final boolean TEZ_AM_REGISTRY_ENABLE_COMPUTE_GROUPS_DEFAULT = false;
+
+
+  /**
+   * Integer value. Initial backoff sleep duration (in milliseconds) for Curator retries.
+   * It's used when creating a CuratorFramework for the ZooKeeper registry.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty
+  public static final String TEZ_AM_CURATOR_BACKOFF_SLEEP = TEZ_AM_PREFIX + "curator.backoff.sleep";
+  public static final int TEZ_AM_CURATOR_BACKOFF_SLEEP_DEFAULT = 1000;
+
+  /**
+   * Integer value. Maximum number of retries for Curator operations.
+   * It's used when creating a CuratorFramework for the ZooKeeper registry.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty
+  public static final String TEZ_AM_CURATOR_MAX_RETRIES = TEZ_AM_PREFIX + "curator.max.retries";
+  public static final int TEZ_AM_CURATOR_MAX_RETRIES_DEFAULT = 3;
+
+  /**
+   * Integer value. Session timeout (in milliseconds) for Curator framework.
+   * It's used when creating a CuratorFramework for the ZooKeeper registry.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty
+  public static final String TEZ_AM_CURATOR_SESSION_TIMEOUT = TEZ_AM_PREFIX + "curator.session.timeout";
+  public static final int TEZ_AM_CURATOR_SESSION_TIMEOUT_DEFAULT = 150000;
```

Comment:
I think we should support time suffix like s ms and all for all these timeout configs